proposals/0089-programify-feature-gate-program.md

```diff
@@ -0,0 +1,149 @@
+---
+simd: '0089'
+title: Programify Feature Gate Program 
+authors:
+  - Joe Caulfield
+category: Standard
+type: Core
+status: Draft
+created: 2023-11-21
+feature: (fill in with feature tracking issues once accepted)
+supersedes: '0077'
+---
+
+## Summary
+
+This proposal suggests replacing the non-existent native program at address
+`Feature111111111111111111111111111111111111` with a Core BPF Program, as
+described in
+[SIMD 0088](https://github.com/solana-foundation/solana-improvement-documents/pull/88).
+
+Feature accounts are already assigned the owner program address
+`Feature111111111111111111111111111111111111`. Deploying a Core BPF program at
+this address would provide engineers with the capability to revoke pending
+feature activations.
+
+**Note:** The process by which core contributors *activate* features would
+remain completely unchanged.
+
+## Motivation
+
+Currently, a feature is queued for activation by a keypair holder creating an
+empty account and assigning it to the
+`Feature111111111111111111111111111111111111` program.
+
+Because there is no actual program implementation at this address, the queuing
+is irreversible; if the runtime knows about a feature gate at some address, it
+will activate it at the next epoch boundary. This means there is no recourse in
+the case of a mistaken queuing, discovery of a bug, or simply a desire to manage
+the cadence and schedule of activations.
+
+A fully-implemented Core BPF program would take ownership of those accounts and
+support revoking queued features, giving engineers more flexibility and
+safeguards.
+
+## Alternatives Considered
+
+The Feature Gate program could instead be implemented as a built-in native
+program, rather than a Core BPF program. However, this would mean any changes to
+the program would need to be implemented by all validator clients in
+coordination. This makes upgrading the program cumbersome and potentially
+dangerous.
```

Comment:
It's not obvious (to me) how upgrading the program is dangerous, per se. I think I'd just remove that part of this sentence.